spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/DefaultOkHttpClientFactory.java

```diff
@@ -0,0 +1,57 @@
+package org.springframework.cloud.commons.httpclient;
+
+import okhttp3.ConnectionPool;
+import okhttp3.OkHttpClient;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeUnit;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Default implementation of {@link OkHttpClientFactory}.
+ * @author Ryan Baxter
+ */
+public class DefaultOkHttpClientFactory implements OkHttpClientFactory {
+
+	private static final Log LOG = LogFactory.getLog(DefaultOkHttpClientFactory.class);
+
+	@Override
+	public OkHttpClient create(boolean disableSslValidation, long connectTimeout,
```

Comment:
I'm a little worried about such a long list of parameters. I wonder if something like a builder customizer might make more sense?